buildSrc/src/main/groovy/com/uber/okbuck/core/model/base/Store.java

```diff
@@ -0,0 +1,73 @@
+package com.uber.okbuck.core.model.base;
+
+import com.uber.okbuck.core.util.FileUtil;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A persistent data store for simple string key/value pairs
+ * that auto cleans up entries that are no longer accessed.
+ */
+public final class Store {
+
+    private final Set<String> accessed;
+    private final File storeFile;
+
+    public final Properties props;
+
+    public Store(File storeFile) {
+        this.storeFile = storeFile;
+        props = new Properties();
+        accessed = ConcurrentHashMap.newKeySet();
+
+        if (storeFile.exists()) {
+            try {
+                props.load(new BufferedInputStream(new FileInputStream(storeFile)));
+            } catch (IOException ignored) { }
```

Comment:
There is no danger if it does not load as okbuck will treat it as a fresh build and will just compute metadata again, which is why it is ignored here